pkg/util/mount/mount_helper_windows.go

```diff
@@ -0,0 +1,43 @@
+// +build windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mount
+
+import (
+	"k8s.io/klog"
+	"strings"
+)
+
+const (
+	// SMBConnectionError : SMB Connection Error, e.g. password change
+	SMBConnectionError = "password is incorrect"
+)
+
+// IsCorruptedMnt return true if err is about corrupted mount point
+func IsCorruptedMnt(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	if strings.Contains(err.Error(), SMBConnectionError) {
```

Comment:
Is "password is incorrect" the only case for corrupt? Are there any other possibilities?

Reply:
good question, there must be other possiblities, while I don't know.
another fix is write like this which may have drawbacks, it depens on whether we use whiltelist or backlist:
```
func IsCorruptedMnt(err error) bool {
    return err != nil
}
```

I think there are some errors still missing on Linux implementation since only following 4 errors caught in `IsCorruptedMnt` func:
https://github.com/kubernetes/kubernetes/blob/a8492d74ecacea9503f08b8e6bf664777b7d41e4/pkg/util/mount/mount_helper.go#L107-L123

@jingxu97 @msau42 @davidz627 what do you think about this? why way do you prefer on Windows?

Reply:
invalid password seems like a strange scenario to consider as a "corrupted" mount.  Is that a normal error that users may encounter?  Is it recoverable?

Reply:
I think these could also be returned
- ERROR_NETWORK_ACCESS_DENIED
- ERROR_BAD_NET_NAME
- ERROR_NETNAME_DELETED

(reference list: https://docs.microsoft.com/en-us/windows/desktop/Debug/system-error-codes--0-499-)

Reply:
@andyzhangx 

> I think there are some errors still missing on Linux implementation since only following 4 errors caught in IsCorruptedMnt func

this command uses Stat() -> fstatat() on *unix
and i think the list of existing errors is odd.

here is what fstatat returns:
http://man7.org/linux/man-pages/man3/fstatat.3p.html

----

on Windows Stat() uses the stat() function here:
https://golang.org/src/os/stat_windows.go

so it will throw a `PathError` at least. i don't know where SMBConnectionError comes from.

in terms of Windows error codes @PatrickLang already listed some, i don't know the codes of a corrupted `mklink / CreateSymbolicLink`, but here is what you can do to check against them:

```
package main

import (
	"fmt"
	"golang.org/x/sys/windows"
	"syscall"
)

func main() {
	ptr, err := windows.UTF16PtrFromString("./some-missing-path")
	if err != nil {
		fmt.Println(err)
		return
	}
	attrib, err := windows.GetFileAttributes(ptr)
	if err != nil {
		fmt.Println(err.(syscall.Errno) == windows.ERROR_FILE_NOT_FOUND)
		return
	}
	fmt.Println(attrib)
}
```

Reply:
@neolit123 thanks for the check, it will throw a `PathError` error, the return error is like following by invoking Lstat --> GetFileAttributesEx, and then error info `The user name or password is incorrect` does not exists in `types_windows.go`:
```
Lstat: GetFileAttributesEx \\f2fd4bb983fba11e9bc4200.file.core.windows.net\andy-k8swin1126-dynami-pvc-2fc7d669-3fba-11e9-8003-000d3a0d9b61: The user name or password is incorrect.
```

Reply:
> and then error info The user name or password is incorrect does not exists in types_windows.go

we can still verify against the integer (ERROR_LOGON_FAILURE, 1326)
https://docs.microsoft.com/en-us/windows/desktop/debug/system-error-codes--1300-1699-

e.g.
```
const ERROR_LOGON_FAILURE = 1326
... 
err.(syscall.Errno) == ERROR_LOGON_FAILURE
```

Reply:
@neolit123 thanks, it's `1040`, not in the list of https://docs.microsoft.com/en-us/windows/desktop/Debug/system-error-codes--1000-1299-, so shall I go with `1040` error checking?

Reply:
hm, i don't know what 1040 is, but probably best not to check for it.
golang just calls native functions and those follow only the existing error codes.

Reply:
@neolit123 sorry, I was wrong, it's this error:
```
ERROR_LOGON_FAILURE
1326 (0x52E)
The user name or password is incorrect.
```
I think I have the correct solution now, I have this error check in the code first, thanks!

Reply:
that's great that it matches the real error codes from windows